tornado/tcpclient.py

```diff
@@ -93,7 +94,13 @@ def try_connect(self, addrs):
                 self.future.set_exception(self.last_error or
                                           IOError("connection failed"))
             return
-        future = self.connect(af, addr)
+        try:
+            future = self.connect(af, addr)
+        except socket.error as e:
```

Comment:
I think this needs to be in `on_connect_done` so it is consistent with the rest of our `connect()` error handling.